src/tests/ut/reserve_queue/reserve_queue_test.cpp

```diff
@@ -0,0 +1,412 @@
+    Treatment filling_treatment{Treatment::DentalFilling};
+    sut_.addAwaitingPatient(filling_treatment, patients_[0]);
+    sut_.addAwaitingPatient(filling_treatment, patients_[1]);
+    sut_.addAwaitingPatient(filling_treatment, patients_[2]);
+    sut_.addAwaitingPatient(filling_treatment, patients_[3]);
+
+    Treatment tooth_extraction_treatment{Treatment::ToothExtraction};
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[3]);
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[0]);
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[1]);
+
+    const auto filling_treatment_queued_patients = sut_.awaitingPatients(filling_treatment);
+    const auto tooth_extraction_queed_patients = sut_.awaitingPatients(tooth_extraction_treatment);
+
+    EXPECT_THAT(filling_treatment_queued_patients, ElementsAre(patients_[0], patients_[1], patients_[2], patients_[3]));
+    EXPECT_THAT(tooth_extraction_queed_patients, ElementsAre(patients_[3], patients_[0], patients_[1]));
+}
+
+TEST_F(AddAwaitingPatientShould, NotEnQueuePatientsIfAlreadyEnqueuedForSpecificTreatment)
+{
+    Treatment tooth_extraction_treatment{Treatment::ToothExtraction};
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[3]);
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[0]);
+
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[3]);
+    sut_.addAwaitingPatient(tooth_extraction_treatment, patients_[0]);
+
+    const auto tooth_extraction_queed_patients = sut_.awaitingPatients(tooth_extraction_treatment);
+
+    EXPECT_THAT(tooth_extraction_queed_patients, ElementsAre(patients_[3], patients_[0]));
+}
+
+TEST_F(AddAwaitingPatientShould, EnqueuePatientsForDoctorInCorrectOrder)
+{
+    // first doctor queue
+    sut_.addAwaitingPatient(doctors_[0], patients_[3]);
+    sut_.addAwaitingPatient(doctors_[0], patients_[2]);
+    sut_.addAwaitingPatient(doctors_[0], patients_[1]);
+    sut_.addAwaitingPatient(doctors_[0], patients_[0]);
+    // second doctor queue
+    sut_.addAwaitingPatient(doctors_[1], patients_[2]);
+    sut_.addAwaitingPatient(doctors_[1], patients_[0]);
+    // third doctor queue
+    sut_.addAwaitingPatient(doctors_[2], patients_[1]);
+    sut_.addAwaitingPatient(doctors_[2], patients_[0]);
+    sut_.addAwaitingPatient(doctors_[2], patients_[3]);
+    sut_.addAwaitingPatient(doctors_[2], patients_[2]);
+
+    const auto first_doctor_queued_patients = sut_.awaitingPatients(doctors_[0]);
+    const auto second_doctor_queued_patients = sut_.awaitingPatients(doctors_[1]);
+    const auto third_doctor_queued_patients = sut_.awaitingPatients(doctors_[2]);
+
+    EXPECT_THAT(first_doctor_queued_patients, ElementsAre(patients_[3], patients_[2], patients_[1], patients_[0]));
+    EXPECT_THAT(second_doctor_queued_patients, ElementsAre(patients_[2], patients_[0]));
+    EXPECT_THAT(third_doctor_queued_patients, ElementsAre(patients_[1], patients_[0], patients_[3], patients_[2]));
+}
```

Comment:
similarly here, do we need that many objects? 
Two doctors would be enough, also number could be reduced

could be reduced in next tests also

Reply:
DONE
(number of patients reduced to 2 in further tests as well)